File: rinhaback2401/Counters.cs
using RinhaBack2401.Model;
using System.Diagnostics.Metrics;

namespace RinhaBack2401;

public static class MeterExtensions
{
    public static void AddCustomMeters(this WebApplication app)
    {
        var db = app.Services.GetRequiredService<Db>();
        var meterFactory = app.Services.GetRequiredService<IMeterFactory>();
        var meter = meterFactory.Create("Rinha", "1.0.0");
        meter.CreateObservableGauge("Connections Available", () => db.QuantityConnectionPoolItemsAvailable);
        meter.CreateObservableGauge("Command (Insert) Available", () => db.QuantityInsertCommandPoolItemssAvailable);
        meter.CreateObservableGauge("Command (Cliente) Available", () => db.QuantityGetClienteCommandPoolItemssAvailable);
        meter.CreateObservableGauge("Command (Transações) Available", () => db.QuantityGetTransacoesCommandPoolItemssAvailable);
        meter.CreateObservableGauge("Connections waiting for a pool item", () => db.QuantityConnectionPoolItemsWaiting);
        meter.CreateObservableGauge("Command (Insert) waiting for a pool item", () => db.QuantityInsertCommandPoolItemssWaiting);
        meter.CreateObservableGauge("Command (Cliente) waiting for a pool item", () => db.QuantityGetClienteCommandPoolItemssWaiting);
        meter.CreateObservableGauge("Command (Transações) waiting for a pool item", () => db.QuantityGetTransacoesCommandPoolItemssWaiting);
    }
}

File: rinhaback2401vb/Counters.vb
Imports Microsoft.AspNetCore.Builder
Imports Microsoft.Extensions.DependencyInjection
Imports RinhaBack2401.Model
Imports System.Diagnostics.Metrics
Imports System.Runtime.CompilerServices

Module MeterExtensions
    <Extension()>
    Sub AddCustomMeters(ByVal app As WebApplication)
        Dim db = app.Services.GetRequiredService(Of Db)()
        Dim meterFactory = app.Services.GetRequiredService(Of IMeterFactory)()
        Dim meter = meterFactory.Create("Rinha", "1.0.0")
        meter.CreateObservableGauge("Command (Insert) Available", Function() db.QuantityInsertCommandPoolItemssAvailable)
        meter.CreateObservableGauge("Command (Cliente) Available", Function() db.QuantityGetClienteCommandPoolItemssAvailable)
        meter.CreateObservableGauge("Command (Transações) Available", Function() db.QuantityGetTransacoesCommandPoolItemssAvailable)
        meter.CreateObservableGauge("Command (Insert) waiting for a pool item", Function() db.QuantityInsertCommandPoolItemssWaiting)
        meter.CreateObservableGauge("Command (Cliente) waiting for a pool item", Function() db.QuantityGetClienteCommandPoolItemssWaiting)
        meter.CreateObservableGauge("Command (Transações) waiting for a pool item", Function() db.QuantityGetTransacoesCommandPoolItemssWaiting)
    End Sub
End Module


